# Nteflix Content Analysis
**Analyst:** Kushagra yadav

**Date:** July 2026

**Dataset:** 8,807 Netflix titles | 2008-2021

## Objective
To analyze Netlifx's content library and uncover key insights around content type distribution, country-wise production, genre trends, and yearly growth - with actionable recommendations for content strategy.

## Tools Used
Python (Pandas, Matplotlib, Seaborn) | SQL(SQLite) | Power BI | Excel | Git

---
## Importing Libraries and Loading dataset

In [10]:
import pandas as pd
import numpy as np

df = pd.read_csv("netflix_titles.csv")

print("Shape:", df.shape)
print("\nColumns:",list(df.columns))
print("\nData Types:")
print(df.dtypes)
df.head()

Shape: (8807, 12)

Columns: ['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added', 'release_year', 'rating', 'duration', 'listed_in', 'description']

Data Types:
show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object


,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...


---
### Checking for the missing values

In [17]:
print("Missing values:")
print(df.isnull().sum())

#missing value percentage
print("\nMissing percentage:")
print(round(df.isnull().sum()/len(df)*100, 2))

Missing values:
show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

Missing percentage:
show_id          0.00
type             0.00
title            0.00
director        29.91
cast             9.37
country          9.44
date_added       0.11
release_year     0.00
rating           0.05
duration         0.03
listed_in        0.00
description      0.00
dtype: float64


---
## Checking the content type breakdown

In [26]:
print("Content Type Breakdown:")
print(df['type'].value_counts())

print("\nPercentage:")
print(round(df['type'].value_counts()/len(df)*100, 2))

Content Type Breakdown:
type
Movie      6131
TV Show    2676
Name: count, dtype: int64

Percentage:
type
Movie      69.62
TV Show    30.38
Name: count, dtype: float64


---
## Checking year range

In [25]:
print("Year Range:")
print("Oldest content:", df['release_year'].min())
print("Newest content:", df['release_year'].max())

print("\nContent added to netflix (date_added sample):")
print(df['date_added'].dropna().head(5))

Year Range:
Oldest content: 1925
Newest content: 2021

Content added to netflix (date_added sample):
0    September 25, 2021
1    September 24, 2021
2    September 24, 2021
3    September 24, 2021
4    September 24, 2021
Name: date_added, dtype: object


---

## Business Questions this Project will Answer

1. Which country produces the most Netflix content?
2. What is the ratio of Movies vs TV Shows on Netflix?
3. Which genres dominate the Netflix library?
4. How has Netflix grown its content library year over year - especially during COVID (2019-2021)?
5. What content ratings are most common and what does that say about Netflix's target audience?

---
**1. Why did i choose "fill" for some and "drop rows" for others?**

Some columns(director) is not important because **director** -> 29.91% missing value -> fill "Unknown" -> not used in analysis but kept for completeness.
Some columns(cast, country, rating, duration, date_added) are very important for the analysis and these columns are important because:
- **cast** -> 9% missing -> fill "Unknown" -> not directly used in business questions.
- **country** -> 9% missing -> fill "Unknown" -> *CRITICAL* — needed for top countries.
- **date_added** -> 0.11% missing -> drop those 10 rows -> tiny amount, won't affect analysis.
- **rating** -> 0.05% missing -> drop those 4 rows -> tiny amount, won't affect analysis.
- **duration** -> 0.03% missing -> drop those 3 rows -> tiny amount, won't affect analysis.


**2. For each null column - why did i choose fill vs drop? What percentage threshold makes dropping rows acceptable vs unacceptable?**

**NULL HANDLING STRATEGY**
- Rule 1: If null % is above 5% AND column is important -> fill with "Unknown"
- Rule 2: If null % is below 1% -> drop those rows safely (minimal data loss)
- Rule 3: If column is not used in any business question -> fill "Unknown" and ignore

**Applied to this dataset:**
- director  -> 29.91% missing -> fill "Unknown" (not used in analysis)
- cast      -> 9.37%  missing -> fill "Unknown" (not used in analysis)
- country   -> 9.44%  missing -> fill "Unknown" (used in Q1 - handle carefully)
- date_added-> 0.11%  missing -> drop 10 rows  (safe - only 10 rows lost)
- rating    -> 0.05%  missing -> drop 4 rows   (safe - only 4 rows lost)
- duration  -> 0.03%  missing -> drop 3 rows   (safe - only 3 rows lost)
---

# Data Cleaning

In [19]:
import pandas as pd
import numpy as np

df = pd.read_excel("netflix_titles.xlsc")
print("Loaded successfully:", df.shape)

FileNotFoundError: [Errno 2] No such file or directory: 'netflix_titles.xlsc'